pkgs/development/yottadb/default.nix

```diff
@@ -0,0 +1,192 @@
+{
+  lib,
+  makeWrapper,
+  stdenv,
+  writeText,
+  writeScript,
+
+  cmake,
+  coreutils-full,
+  elfutils,
+  fetchFromGitLab,
+  git,
+  glibcLocales,
+  icu,
+  ncurses,
+  openssl,
+  pkg-config,
+  tcsh,
+  util-linux,
+  zlib
+}:
+
+stdenv.mkDerivation rec {
+  pname = "yottadb";
+  version = "1.34";
+
+  src = fetchFromGitLab {
+    owner = "YottaDB";
+    repo = "DB/YDB";
+    rev = "r${version}";
+    leaveDotGit = true;
+    sha256 = "sha256-rYKw+R2VF+aht4UTcsiGVavIbmTRwpuUGHQe8LJCFQo=";
+  };
+
+  enableParallelBuilding = true;
+
+  nativeBuildInputs = buildInputs ++ [
+    cmake
+    git
+    icu.dev
+    makeWrapper
+    ncurses
+    pkg-config
+    tcsh
+    zlib
+  ];
+
+  buildInputs = [
+    coreutils-full
+    elfutils.dev
+    glibcLocales
+    icu
+    openssl
+    util-linux
+  ];
+
+  # The version number is generated using git-describe
+  # and any code patching makes this version number dirty.
+  # Do not modify the git repo in any way!
+  dontFixCmake = true;
+
+  preConfigure = ''
+    # The following cannot be passed to cmakeFlags directly due
+    # to variable substitution
+    cmakeFlags="$cmakeFlags -DYDB_INSTALL_DIR:STRING=$out/dist"
+  '';
+
+  preBuild = ''
+    # Make sure that we can find all the runtime libs in order to
+    # use the freshly built M compiler to build the M code...
+    export LD_LIBRARY_PATH="${lib.makeLibraryPath buildInputs}:$LD_LIBRARY_PATH"
+
+    # Revert any potential repo modifications to keep the release clean
+    git reset --hard
+  '';
+
+  # NOTE: Setting exec_prefix=@@YDB_EXEDIR@@, includedir=@@YDB_INCDIR@@
+  # and libdir=@@YDB_LIBDIR@@ breaks dependent software (YDBRust for
+  # instance) for some reason.
+  yottadbPkgConfigTemplate = writeText "yottadb.pc.in" ''
+    prefix=@@YDB_DIST@@
+
+    exec_prefix=''${prefix}
+    includedir=''${prefix}
+    libdir=''${exec_prefix}
+
+    Name: YottaDB
+    Description: YottaDB database library
+    Version: ${version}
+    Cflags: -I''${includedir}
+    Libs: -L''${libdir} -lyottadb -Wl,-rpath,''${libdir}
+  '';
+
+  yottadbGde = writeScript "gde" ''
+    #!/usr/bin/env bash
+    exec ''${0%/*}/yottadb -r GDE "$@"
+  '';
+
+  postInstall = ''
+    mkdir -p $out/bin $out/lib $out/dist/utf8
+    mkdir -p $out/lib/pkgconfig $out/include
+
+    # We cannot set SUID for gtmsecshr, it has to be done using Nix Security Wrapper
+    cd $out/dist
+    mv gtmsecshr gtmsecshr.nosuid
+    ln -rfs /run/wrappers/bin/gtmsecshr .
```

Comment:
@ajs124 is there any better way of doing this? This simply assumes that the wrapper is there (module enabled).

I can only think of creating yet another wrapper (to replace this symlink) that would detect if the SUID wrapper is available and either call it or fail. Is there any other way?